### PR TITLE
Native iOS Panel

### PR DESCRIPTION
**Pull request by k0nserv**

### CODE REVIEW

File: native/packages/react-native-backpack/libraries/Panel/ios/react/classes/BPKPanelWrapper.m [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+- (void)setup {
+    BPKPanel *panel = [[BPKPanel alloc] initWithFrame:self.bounds];
+    self.inner = panel;
+
+    [self addSubview:panel];
+    panel.translatesAutoresizingMaskIntoConstraints = NO;
+    [panel.layoutMarginsGuide.leadingAnchor constraintEqualToAnchor:self.leadingAnchor].active = YES;
+    [panel.layoutMarginsGuide.trailingAnchor constraintEqualToAnchor:self.trailingAnchor].active = YES;
+    [panel.layoutMarginsGuide.topAnchor constraintEqualToAnchor:self.topAnchor].active = YES;
+    [panel.layoutMarginsGuide.bottomAnchor constraintEqualToAnchor:self.bottomAnchor].active = YES;
```
Comment (shaundon) [inline native/packages/react-native-backpack/libraries/Panel/ios/react/classes/BPKPanelWrapper.m]: Is this the same as using `Ctrl + drag` to create constraints through Interface Builder?

File: native/packages/react-native-backpack/libraries/Panel/index.js [hunk 2]
```diff
@@ -16,5 +16,30 @@
...
  */
 /* @flow */
+import React, { Component, type Node } from 'react';
+import PropTypes from 'prop-types';
+import { requireNativeComponent } from 'react-native';
 
-// Not much to see here
+type Props = {
+  children: Node,
+  padded: boolean,
```
Comment (shaundon) [inline native/packages/react-native-backpack/libraries/Panel/index.js]: Is `padded` non-functional right now? I tried applying it in the Storybook and through Interface Builder and it doesn't seem to have an effect.

File: native/packages/react-native-backpack/libraries/Panel/ios/react/classes/BPKPanelWrapper.m (hunk 1 above)
Comment (k0nserv) [reply native/packages/react-native-backpack/libraries/Panel/ios/react/classes/BPKPanelWrapper.m]: Yup more or less

File: native/packages/react-native-backpack/libraries/Panel/index.js (hunk 2 above)
Comment (k0nserv) [reply native/packages/react-native-backpack/libraries/Panel/index.js]: Yeah pretty much, the React Native part of this implementation doesn't really work.

**Comment by k0nserv** [comment]
Very good point about the storybook. Will have to think of something

**Comment by shaundon** [comment]
Other than the Storybook thing we discussed, I think this looks good, based on my still rudimentary iOS knowledge (although a week ago I'd have struggled to review this at all!). Only other things are:

* I think we should document somewhere (readme, comments in code, somewhere else) that this is currently non-functional. I know that **we** know, but if we're merging non-working code into `master` I think that's a special case to be noted for anybody else viewing this repo.

* `eslint` is broken, you just need to add some opt-outs.

**Comment by k0nserv** [comment]
There's a big warning in the `README.md` already which states this is all WIP. Do you think that is enough or would you want to add something more? Will fix `eslint`

File: native/ios/native/ToggleNavigationBarMenuModule.h [hunk 3]
```diff
@@ -0,0 +1,15 @@
+//
+//  ToggleNavigationBarMenuModule.h
+//  Backpack Native
+//
+//  Created by Hugo Tunius on 01/08/2018.
+//  Copyright © 2018 Facebook. All rights reserved.
+//
```
Comment (shaundon) [inline native/ios/native/ToggleNavigationBarMenuModule.h]: 🙈 Is there something you aren't telling us? 😉

**Pull request merged**